compiler/rustc_data_structures/src/obligation_forest/mod.rs

```diff
@@ -251,12 +251,21 @@ enum NodeState {
     Error,
 }
 
+/// This trait allows us to make Outcome an opaque type, allowing a different implementation
+/// for tests.
```

Comment:
I didn't understand this comment at first. Maybe change it to this: "This trait allows us to have two implementations of `Outcome`: the normal one, and one that does some additional work and checking for use in tests."

Reply:
Thanks, I've updated this.